index.d.mts

```diff
@@ -0,0 +1,50 @@
+import * as i18next from './index.js';
+
+import type { DefaultNamespace, Namespace } from './typescript/options.js';
+
+export type WithT<Ns extends Namespace = DefaultNamespace> = i18next.WithT<Ns>;
```

Comment:
If you want, we can move these common typings into a separate `common.d.ts` file.

Reply:
Right, that's probably a good idea

Reply:
@perrin4869 Interesting, the following would error:

```ts
// Internal Helpers
import type { WithT, i18n } from './typescript/common.js';

declare namespace i18next {
  export { WithT, i18n };
}

declare const i18next: i18n;

export = i18next;
```

<img width="508" alt="image" src="https://github.com/i18next/i18next/assets/8336744/2ef8a9eb-7c52-492e-ba02-febd0ff4d99d">

[Playground](https://www.typescriptlang.org/play?target=99&jsx=1&module=1#code/JYWwDg9gTgLgBDAnmApnA3nA6sGALAFQBo5gBGADgDs4BfOAMyghDgHIA6AeiVQGcAxlGBgYXASxAQqHAFZ82AbgBQygCYoBAGwCGUNFR0gUfMDoFpy1FAA946ZXDi3IsDNlyESVmrRW1VDW09NAkqPngfWxgALlJKKhVlF2h4AF546ztFIA)

Any idea how to fix this to simplify?

Reply:
oh damn that's interesting... doesn't seem like that's going to work out...